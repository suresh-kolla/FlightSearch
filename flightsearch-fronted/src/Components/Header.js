import React, { Component } from "react";
import "./../styles/Header.css";

export default class Header extends Component {
  render() {
    return (
      <div className="App-header" data-test="component-header">
        <h1 className="App-title" data-test="header-word">
          {this.props.title}
        </h1>
      </div>
    );
  }
}
