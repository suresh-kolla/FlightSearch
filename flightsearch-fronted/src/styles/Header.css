.App-header {
  background-color: #222;
  height: 150px;
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
}
