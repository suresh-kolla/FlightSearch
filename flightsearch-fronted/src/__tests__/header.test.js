import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { findByTestAttr, checkProps } from "./testUtils";
import Header from "./../Components/Header";

const setup = (props = {}) => {
  return shallow(<Header {...props} />);
};

test("renders with no error", () => {
  const wrapper = setup();
  const component = findByTestAttr(wrapper, "component-header");
  expect(component.length).toBe(1);
});

test("renders Header with out empty", () => {
  const wrapper = setup({ title: "Flight Search" });
  const header = findByTestAttr(wrapper, "header-word");
  expect(header.text().length).not.toBe(0);
});
