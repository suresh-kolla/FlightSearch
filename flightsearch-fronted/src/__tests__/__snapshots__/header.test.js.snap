// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<div
  className="App-header undefined"
>
  <h1
    color="text-white"
  >
    Facebook
  </h1>
</div>
`;
